components/ButtonControl.js

```diff
@@ -1,34 +1,34 @@
-import React, { Component } from 'react';
+import React, { Component } from "react";
 import { ButtonGroup } from "react-native-elements";
-import {Text} from "react-native";
+import { Text } from "react-native";
 
 class ButtonControl extends Component<Props> {
-    constructor() {
-        super()
-        this.state = {
-            selectedIndex: 0
-        }
-        this.updateIndex = this.updateIndex.bind(this)
-    }
+  constructor() {
+    super();
+    this.state = {
+      selectedIndex: 0
+    };
+    this.updateIndex = this.updateIndex.bind(this);
+  }
 
-    updateIndex(selectedIndex) {
-        this.setState({ selectedIndex })
-        this.props.onChangeIndex(selectedIndex)
-    }
+  updateIndex(selectedIndex) {
+    this.setState({ selectedIndex });
+    this.props.onChangeIndex(selectedIndex, this.props.starredItems);
```

Comment:
do you think the selectedIndex should just be passed down along with a callback to change index? Doesn't seem to be a reason to hold state heres

Reply:
tru tru